spec/models/reports/repository_spec.rb

```diff
@@ -112,7 +120,10 @@ def refresh_views
 
     it "gets controlled counts and rates for single region" do
       facilities = FactoryBot.create_list(:facility, 2, facility_group: facility_group_1).sort_by(&:slug)
+      other_facility = create(:facility)
       facility_1, facility_2 = facilities.take(2)
+      # TODO we shouldn't need to create a registered patient here to make FacilityStates return records...
+      _facility_1_registered_but_not_assigned = create(:patient, recorded_at: jan_2019, assigned_facility: other_facility, registration_facility: facility_1, registration_user: user)
```

Comment:
Here was one other problem w/ the JOINs we have -- we need a registered patient early enough w/ a facility for the left join to work.  Maybe we could use a `FULL OUTER JOIN` across registered, assigned, and treatment outcomes to get all the things regardless of what happens first for a facility?  

Thinking thru a real world case of this: a facility starts at the end of the month, some patients get assigned at month end, and they don't get new registrations until the start of the next month.  The assigned numbers for the end of month would not be counted until registrations start happening.

Reply:
https://github.com/simpledotorg/simple-server/pull/2734#discussion_r672877926